src/pages/fi-receipt-vouchers/forms/ReceiptVoucherForm.js

```diff
@@ -0,0 +1,497 @@
+      cashAccountId: null,
+      plantId: null,
+      exRate: 1.0,
+      rateCalcMethod: 1,
+      contactId: null,
+      collectorId: null,
+      isVerified: true,
+      template: 1
+    },
+    validationSchema: yup.object({
+      accountId: yup.string().required(' '),
+      currencyId: yup.string().required(' '),
+      cashAccountId: yup.string().required(' '),
+      amount: yup.string().required(' '),
+      paymentMethod: yup.string().required(' ')
+    }),
+    onSubmit: async obj => {
+      try {
+        const response = await postRequest({
+          extension: FinancialRepository.ReceiptVouchers.set,
+          record: JSON.stringify(obj)
+        })
+        if (!obj.recordId) {
+          toast.success('Record Added Successfully')
+          formik.setFieldValue('recordId', response.recordId)
+          getData(response.recordId)
+        } else toast.success('Record Edited Successfully')
+        invalidate()
+      } catch (e) {}
+    }
+  })
+  const editMode = !!recordId || !!formik.values.recordId
+
+  const getDefaultDT = async () => {
+    const userData = getStorageData('userData')
+
+    const _userId = userData.userId
+    try {
+      const { record: cashAccountRecord } = await getRequest({
+        extension: SystemRepository.UserDefaults.get,
+        parameters: `_userId=${_userId}&_key=cashAccountId`
+      })
+
+      const { record: currency } = await getRequest({
+        extension: SystemRepository.Default.get,
+        parameters: `_key=currencyId`
+      })
+      const cashAccountId = cashAccountRecord?.value
+      if (cashAccountId) {
+        const { record: cashAccountResult } = await getRequest({
+          extension: CashBankRepository.CbBankAccounts.get,
+          parameters: `_recordId=${cashAccountId}`
+        })
+
+        formik.setFieldValue('cashAccountId', cashAccountId)
+        formik.setFieldValue('cashAccountRef', cashAccountResult.reference)
+        formik.setFieldValue('cashAccountName', cashAccountResult.name)
+        formik.setFieldValue('currencyId', parseInt(currency.value))
+      }
+
+      const { record: plantRecord } = await getRequest({
+        extension: SystemRepository.UserDefaults.get,
+        parameters: `_userId=${_userId}&_key=plantId`
+      })
+      if (plantRecord) {
+        formik.setFieldValue('plantId', parseInt(plantRecord.value))
+      }
+    } catch (error) {}
+  }
+
+  useEffect(() => {
+    formik.setFieldValue('templateId', '')
+  }, [formik.values.notes])
+
+  useEffect(() => {
+    !recordId && getDefaultDT()
+    ;(async function () {
+      getData(recordId)
+    })()
+  }, [])
+
+  async function getData(_recordId) {
+    try {
+      const recordId = _recordId || recordId
+      if (recordId) {
+        const res = await getRequest({
+          extension: FinancialRepository.ReceiptVouchers.get,
+          parameters: `_recordId=${recordId}`
+        })
+        setIsPosted(res.record.status === 3 ? true : false)
```

Comment:
Operations on records shouldn't be done like that. Please consult @Ghinwa-Arab  for a previous discussion we had regarding this. The whole record should be reloaded.